File: tundra-cli/benchmarks/sieve/sieve.php
<?php
// File: sieve/sieve.php
// Sieve of Eratosthenes in PHP

function sieve($n) {
    $is_composite = array_fill(0, $n + 1, false);
    $p = 2;
    while ($p * $p <= $n) {
        if (!$is_composite[$p]) {
            for ($m = $p * $p; $m <= $n; $m += $p) {
                $is_composite[$m] = true;
            }
        }
        $p++;
    }
    $count = 0;
    for ($i = 2; $i <= $n; $i++) {
        if (!$is_composite[$i]) {
            $count++;
        }
    }
    echo $count . "\n";
}

$n=200000;
sieve($n);

File: tundra-cli/benchmarks/sieve/sieve.rb
# File: sieve/sieve.rb
# Sieve of Eratosthenes in Ruby

def sieve(n)
  is_composite = Array.new(n + 1, false)
  p = 2
  while p * p <= n
    unless is_composite[p]
      (p * p).step(n, p) { |m| is_composite[m] = true }
    end
    p += 1
  end
  count = 0
  (2..n).each { |i| count += 1 unless is_composite[i] }
  puts count
end

n =200000
sieve(n)
